internal/replacements/replacements_test.go

```diff
@@ -501,14 +527,20 @@ var _ = Describe("replace_misconfigured_pods", func() {
 					JustBeforeEach(func() {
 						newCPU, err := resource.ParseQuantity("1000")
 						Expect(err).NotTo(HaveOccurred())
-						cluster.Spec.Processes[fdbv1beta2.ProcessClassGeneral].PodTemplate.Spec.Containers = append(cluster.Spec.Processes[fdbv1beta2.ProcessClassGeneral].PodTemplate.Spec.Containers,
+
+						setting, present := cluster.GetBareProcessSettings(fdbv1beta2.ProcessClassGeneral)
+						Expect(present).To(BeTrue())
+
+						setting.PodTemplate.Spec.Containers = append(setting.PodTemplate.Spec.Containers,
 							corev1.Container{
 								Resources: corev1.ResourceRequirements{
 									Requests: corev1.ResourceList{
 										corev1.ResourceCPU: newCPU,
 									},
 								},
 							})
+
+						cluster.UpdateBareProcessSettings(fdbv1beta2.ProcessClassGeneral, setting)
```

Comment:
If we're repeating this fetch-and-update pattern multiple times, could we can encapsulate it in a function that takes a callback?